File: c/LeetCode_189_RotateArray.c
#include <stdio.h>
#include <stdlib.h>


void rotate(int* nums, int numsSize, int k) {
    k = numsSize - (k % numsSize);

    int inx = 0;
    int *tmp = (int*) malloc(numsSize * sizeof(int));
    for (int i = k; i < numsSize; ++i) {
        tmp[inx] = nums[i];
        ++inx;
    }

    for (int i = 0; i < k; ++i) {
        tmp[inx] = nums[i];
        ++inx; 
    }

    for (int i = 0; i < numsSize; ++i) {
        nums[i] = tmp[i];
    }

    free(tmp);
}


int main() {
    return 0;
}

File: python/LeetCode_189_RotateArray.py
def rotate(nums, k):
    length = len(nums)

    k = length - (k % length)
    tmp = [0 for _ in range(length)]

    inx = 0
    for i in range(k, length):
        tmp[inx] = nums[i]
        inx += 1

    for i in range(k):
        tmp[inx] = nums[i]
        inx += 1

    for i in range(length):
        nums[i] = tmp[i]


rotate([1,2,3,4,5,6,7], 6)



"""
def rotate(nums, k: int) -> None:
    tmp = 0 
    k = k % len(nums)
    for _ in range(k):
        tmp = nums.pop()
        nums.insert(0, tmp)
"""